Redirect after confirming account

orangejenny opened the pull request:
http://manage.dimagi.com/default.asp?176522

On success, redirect to dashboard with a message:
![screen shot 2015-07-15 at 6 24 05 pm](https://cloud.githubusercontent.com/assets/1486591/8711316/c00570b2-2b1e-11e5-8c7b-288d1e5405f9.png)

On error, show the same page as previously:
![screen shot 2015-07-15 at 6 25 28 pm](https://cloud.githubusercontent.com/assets/1486591/8711334/f4130842-2b1e-11e5-9ed6-68e2030cb0ae.png)

@czue / @nickpell

orangejenny committed:
Redirect after confirming account

dimagimon commented on corehq/apps/registration/views.py:
```diff
@@ -310,18 +304,12 @@ def confirm_domain(request, guid=None):
-    ) % requesting_user.username
-    context['is_error'] = False
-    context['valid_confirmation'] = True
+    messages.success(request, 
```
W291 trailing whitespace

orangejenny commented on corehq/apps/registration/templates/registration/confirmation_complete.html:
```diff
@@ -1,34 +0,0 @@
-<script>
-    {% if not is_error %}
-    $(function () {
-        analytics.workflow("Confirmed new project");
```
@amsagoff note that we're losing this bit of analytics

czue replied:
if we wanted we could push this to python, right? not sure how much of a priority is.

orangejenny replied:
yes, fair point, added them back in server-side

orangejenny committed:
trailing whitespace

czue commented on corehq/apps/registration/templates/registration/confirmation_error.html:
```diff
@@ -0,0 +1,13 @@
+{% block registration-content %}
+    {% if message_body %}<p>{{ message_body }}</p>{% endif %}
+    <p>You may try <a href="{% url "registration_resend_domain_confirmation" %}">resending the confirmation email</a>.</p>
+    <p>In the meantime, please explore the <a href="/a/public/">CommCare HQ Example Project.</a></p>
```
@amsagoff do we still send people here? i didn't even know this thing still existed

amsagoff commented:
@orangejenny confirming that they will normally still be logged in when they click the confirmation link, so they won't be redirected to the login page most of the time.

The demo project is definitely old and we don't redirect people anymore. Here's the current confirmation page, which directs people to the Exchange instead. I think this is still a fine page on error:

![image](https://cloud.githubusercontent.com/assets/439164/8725785/38ea26a0-2ba4-11e5-9ba3-31947b25afe1.png)

@orangejenny instead of "missing activation key" can we change the language to be more user-facing -- e.g. "Account not activated -- email not yet confirmed"

czue commented:
looks good to me pending @amsagoff's requests. will let @nickpell do final review / merge

orangejenny committed:
update copy

orangejenny committed:
put back analytics

orangejenny commented:
@amsagoff Yeah, that wasn't the best screenshot choice. Most of the time, people will already be logged in and will go the dashboard.

Replaced demo project link with exchange link, updated error text.

orangejenny commented:
@nickpell should be g2g
